components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -0,0 +1,241 @@
+        FederatedAuthenticatorConfig[] federatedAuthenticatorConfigs = new FederatedAuthenticatorConfig[10];
+        when(identityProvider.getFederatedAuthenticatorConfigs()).thenReturn(federatedAuthenticatorConfigs);
+
+        mockStatic(IdentityApplicationManagementUtil.class);
+        when(IdentityApplicationManagementUtil.getFederatedAuthenticator(federatedAuthenticatorConfigs, IdentityApplicationConstants.Authenticator.OIDC.NAME)).thenReturn(federatedAuthenticatorConfig);
+        when(IdentityApplicationManagementUtil.getProperty(any(Property[].class), anyString())).thenCallRealMethod();
+        when(federatedAuthenticatorConfig.getProperties()).thenReturn(properties);
+
+        when(oAuth2AuthorizeReqDTO.getUser()).thenReturn(authenticatedUser);
+        when(authenticatedUser.getAuthenticatedSubjectIdentifier()).thenReturn("User1");
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn((long) 3);
+        mockStatic(OAuth2Util.class);
+        when(oAuthServerConfiguration.getOpenIDConnectIDTokenExpiration()).thenReturn(String.valueOf((int) 78383));
+
+        when(oAuth2AuthorizeReqDTO.getNonce()).thenReturn("Nonce1");
+        LinkedHashSet set1 = new LinkedHashSet(10);
+        when(oAuth2AuthorizeReqDTO.getACRValues()).thenReturn(set1);
+        when(oAuth2AuthorizeReqDTO.getResponseType()).thenReturn("Response1");
+
+        JWSAlgorithm jwsAlgorithm = new JWSAlgorithm("Algorithm8");
+        when(OAuth2Util.mapDigestAlgorithm(any(JWSAlgorithm.class))).thenReturn("SHA-256");
+        when(oAuthServerConfiguration.getIdTokenSignatureAlgorithm()).thenReturn("Algo2");
+        mockStatic(FederatedAuthenticatorConfig.class);
+
+        JWSAlgorithm jwsAlgorithm1 = new JWSAlgorithm("Algo3");
+        when(OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(anyString())).thenReturn(jwsAlgorithm1);
+        DefaultIDTokenBuilder defaultIDTokenBuilder = new DefaultIDTokenBuilder();
+
+        mockStatic(MessageDigest.class);
+        when(MessageDigest.getInstance(String.valueOf(JWSAlgorithm.RS256))).thenReturn(messageDigest);
+        when(oAuth2AuthorizeRespDTO.getAccessToken()).thenReturn("AccessToken");
+        when(oAuth2AuthorizeReqDTO.getConsumerKey()).thenReturn("key1");
+        mockStatic(IdentityConfigParser.class);
+        when(IdentityConfigParser.getInstance()).thenReturn(identityConfigParser);
+        when(oAuthServerConfiguration.getOpenIDConnectCustomClaimsCallbackHandler()).thenReturn(customClaimsCallbackHandler);
+        doNothing().when(customClaimsCallbackHandler).handleCustomClaims(jwtClaimsSet, request);
+
+        when(OAuth2Util.signJWT(any(JWTClaimsSet.class), any(JWSAlgorithm.class), anyString())).thenReturn(jwt);
+        assertNotEquals(defaultIDTokenBuilder.buildIDToken(request, oAuth2AuthorizeRespDTO), "Test passed1");
```

Comment:
It's obvious that IdToken never equals to "Test passed1" string, therefore this assert is not right. Use a different Idtoken to check not equal. Better check Equals rather than not equal. 
Put a proper error message as well for the assert.